packages/edit-post/src/components/sidebar/post-template/form.js

```diff
@@ -25,29 +25,26 @@ export default function PostTemplateForm( { onClose } ) {
 		canCreate,
 		canEdit,
 	} = useSelect( ( select ) => {
+		const { canUser, getEntityRecord, getEntityRecords } =
+			select( coreStore );
 		const editorSettings = select( editorStore ).getEditorSettings();
-		const siteSettings = select( coreStore ).getEntityRecord(
-			'root',
-			'site'
-		);
+		const siteSettings = canUser( 'read', 'settings' )
+			? getEntityRecord( 'root', 'site' )
+			: undefined;
 		const _isPostsPage =
 			select( editorStore ).getCurrentPostId() ===
 			siteSettings?.page_for_posts;
-		const canCreateTemplates = select( coreStore ).canUser(
-			'create',
-			'templates'
-		);
+		const canCreateTemplates = canUser( 'create', 'templates' );
+
 		return {
 			isPostsPage: _isPostsPage,
 			availableTemplates: editorSettings.availableTemplates,
-			fetchedTemplates: select( coreStore ).getEntityRecords(
-				'postType',
-				'wp_template',
-				{
-					post_type: select( editorStore ).getCurrentPostType(),
-					per_page: -1,
-				}
-			),
+			fetchedTemplates: canCreateTemplates
+				? getEntityRecords( 'postType', 'wp_template', {
+						post_type: select( editorStore ).getCurrentPostType(),
+						per_page: -1,
+				  } )
+				: undefined,
```

Comment:
Technically we only want to fetch templates when a user has caps to create them to keep the `options` array in sync. See #32744